test/onnx/test_utility_funs.py

```diff
@@ -614,6 +614,64 @@ def forward(self, x):
         # verify that the model state is preserved
         assert model.training == old_state
 
+    @skipIfUnsupportedMinOpsetVersion(12)
+    def test_local_function(self):
+        class N(torch.nn.Module):
+            def __init__(self, prob):
+                super().__init__()
+                self.dropout = torch.nn.Dropout(prob)
+
+            def forward(self, x):
+                return self.dropout(x)
+
+        class M(torch.nn.Module):
+            def __init__(self, num_layers):
+                super().__init__()
+                self.num_layers = num_layers
+                self.lns = torch.nn.ModuleList([torch.nn.LayerNorm(3, eps=i) for i in range(num_layers)])
+                self.celu1 = torch.nn.CELU(1.0)
+                self.celu2 = torch.nn.CELU(2.0)
+                self.dropout = N(0.5)
+
+            def forward(self, x: torch.Tensor, y: torch.Tensor, z: torch.Tensor) -> torch.Tensor:
+                res1 = self.celu1(x)
+                res2 = self.celu2(y)
+                for ln in self.lns:
+                    z = ln(z)
+                return res1 + res2, self.dropout(z)
+
+        x = torch.randn(2, 3)
+        y = torch.randn(2, 3)
+        z = torch.randn(2, 3)
+
+        f = io.BytesIO()
+        torch.onnx.export(M(3), (x, y, z), f, opset_version=self.opset_version,
+                          export_modules_as_functions=[torch.nn.CELU, torch.nn.Dropout, torch.nn.LayerNorm])
```

Comment:
Different level of modules, yet similarly, since exported in eval mode, neither torch.nn.Dropout nor N will exist in the final exported model. Added more test code to cover this.